stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -2704,7 +2704,7 @@ function _spsetnz_setindex!(A::AbstractSparseMatrixCSC{Tv}, x::Tv,
                                 resize!(nzvalA, nnzA)
                             end
                             r = rowidx:(rowidx+(new_stop-new_ptr))
-                            rowvalA[r] .= I[new_ptr:new_stop]
+                            rowvalA[r] .= _getrowinds(I, new_ptr:new_stop)
```

Comment:
```suggestion
                            rowvalA[r] .= I <: Number ? I : I[new_ptr:new_stop]
```
I feel like this might be more concise? But I'm happy either way

Reply:
Agreed.